app/feature_flags.go

```diff
@@ -15,16 +17,15 @@ import (
 func (s *Server) setupFeatureFlags() {
 	s.featureFlagSynchronizerMutex.Lock()
 	defer s.featureFlagSynchronizerMutex.Unlock()
-	license := s.License()
-	inCloud := license != nil && *license.Features.Cloud
 	splitKey := *s.Config().ServiceSettings.SplitKey
-	syncFeatureFlags := inCloud && splitKey != "" && s.IsLeader()
+	splitConfigured := splitKey != ""
+	syncFeatureFlags := splitConfigured && s.IsLeader()
 
-	s.configStore.PersistFeatures(inCloud)
+	s.configStore.PersistFeatures(splitConfigured)
```

Comment:
@crspeller this is the only part I'm unsure about. Should we only persist them for Cloud still?

Reply:
This issue with persisting outside cloud environments is undesirable config file changes. Since you still need to have split.io configured, and end-users shouldn't have that configured, I think this is probably fine.